bpython/repl.py

```diff
@@ -349,7 +365,7 @@ def clear(self) -> None:
 
 
 class Interaction:
-    def __init__(self, config, statusbar=None):
+    def __init__(self, config: Config, statusbar: Optional[Statusbar] = None):
```

Comment:
Note that the `urwid` frontends passes a different `Statusbar`. This needs be solved differently, e.g. by moving to the respective frontends.

Reply:
@sebastinas what do you mean by moving to the respective frontends here?

Reply:
`Interaction.statusbar` is never used in `bpython/repl.py`. That's an implementation detail of the different repls that leaked into the base repl. If a repl needs that statusbar, they can add the statusbar to their `Interaction`.

Reply:
@sebastinas, I'm not positive this is the direction you wanted me to go, but I modified this to import Statusbar from both `cli` and `urwid` and then typed the parameter like this: `statusbar: Union["cliStatusbar", "urwidStatusbar", None] = None'

Let me know if I'm getting warmer or colder. :)